hasher-matcher-actioner/hmalib/matchers/matchers_base.py

```diff
@@ -260,13 +304,19 @@ def get_metadata_objects_from_match(
     def get_index(self, signal_type: t.Type[SignalType]) -> SignalTypeIndex:
         # If cached, return an index instance for the signal_type. If not, build
         # one, cache and return.
-        if not signal_type in self._cached_indexes:
-            index_cls = INDEX_MAPPING[signal_type]
+        max_custom_threshold = (
+            get_max_threshold_of_active_privacy_groups_for_signal_type(signal_type)
+        )
+        index_cls = get_index_for_signal_type_matching(
+            signal_type, max_custom_threshold
+        )
 
+        if not signal_type in self._cached_indexes or not isinstance(
+            self._cached_indexes[signal_type], index_cls
```

Comment:
AFAIK, self._cached_indexes is only updated by this method itself. Why do the type assertion?

Reply:
Answered in other still open conversation and added comment to file.